File: blog-backend/src/main/java/com/blog/entity/Category.java
package com.blog.entity;

import com.blog.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 分类实体类
 */
@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "category")
public class Category extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false, unique = true, length = 50)
    private String name;

    @Column(name = "description", length = 255)
    private String description;

    @Column(name = "sort_order")
    private Integer sortOrder = 0;
}

File: src/types/category.ts
/**
 * 分类相关类型定义
 */

/**
 * 分类接口
 */
export interface Category {
  id: number
  name: string
  description?: string
  sortOrder?: number
  articleCount?: number
  createdAt: string
}

/**
 * 分类请求参数
 */
export interface CategoryRequest {
  id?: number
  name: string
  description?: string
  sortOrder?: number
}